examples/ACE/utils/NNIAP.jl

```diff
@@ -42,22 +76,73 @@ end
 
 
 # Loss function ################################################################
-function loss(nn, iap, ds, w_e = 1, w_f = 1)
+function gpu_loss(nn, iap, ds, w_e = 1, w_f = 1)
+    _device = gpu
+    nniap = NNIAP(nn, iap)
+    es, es_pred = get_all_energies(ds), get_all_energies(ds, nniap, _device)
+    fs, fs_pred = get_all_forces(ds), get_all_forces(ds, nniap, _device)
+
+    e_error = w_e * Flux.mse(es_pred, es) |> cpu
+    fs = fs|> gpu
+    f_error = w_f * Flux.mse(fs_pred, fs) |> cpu
+    total_error = e_error + f_error
+    return total_error
+end
+ 
+
+function loss(nn, iap, ds, w_e::Real=1, w_f::Real=1)
     nniap = NNIAP(nn, iap)
     es, es_pred = get_all_energies(ds), get_all_energies(ds, nniap)
     fs, fs_pred = get_all_forces(ds), get_all_forces(ds, nniap)
     return w_e * Flux.mse(es_pred, es) + w_f * Flux.mse(fs_pred, fs)
 end
 
 
+function loss(nn, iap,  batch, true_energy, local_descriptors, _device, w_e=1, w_f=1)
+    # nniap = NNIAP(nn, iap)
+    es_pred = sum(sum(nn(local_descriptors)))
+    fs, fs_pred = get_all_forces(batch), get_all_forces(batch, nn , local_descriptors, _device)
+    return w_e * Flux.mse(es_pred, true_energy) + w_f * Flux.mse(fs_pred, fs)
+end
+
+
 # Auxiliary functions ##########################################################
+function get_all_energies(ds::DataSet, nniap::NNIAP, _device)
```

Comment:
It is not necessary to add `_device` to this function, the data transfer to the GPU will be handled by the learn function.

Reply:
This function `get_all_energies(ds::DataSet, nniap::NNIAP, _device)` is not needed anymore.